yt_dlp/extractor/allstar.py

```diff
@@ -0,0 +1,251 @@
+import functools
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    OnDemandPagedList,
+    UnsupportedError,
+    int_or_none,
+    parse_qs,
+    traverse_obj,
+    urljoin,
+)
+
+
+_FIELDS = '''
+    _id
+    clipImageThumb
+    clipLink
+    clipTitle
+    createdDate
+    shareId
+    user { _id }
+    username
+    views'''
+
+_EXTRA_FIELDS = '''
+    clipLength
+    clipSizeBytes'''
+
+_QUERIES = {
+    'clip': '''query ($id: String!) {
+        video: getClip(clipIdentifier: $id) {
+            %s %s
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'montage': '''query ($id: String!) {
+        video: getMontage(clipIdentifier: $id) {
+            %s
+        }
+    }''' % _FIELDS,
+    'Clips': '''query ($page: Int!, $user: String!, $game: Int) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: false, game: $game) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+    'Montages': '''query ($page: Int!, $user: String!) {
+        videos: montages(search: createdDate, page: $page, user: $user) {
+            data { %s }
+        }
+    }''' % _FIELDS,
+    'Mobile Clips': '''query ($page: Int!, $user: String!) {
+        videos: clips(search: createdDate, page: $page, user: $user, mobile: true) {
+            data { %s %s }
+        }
+    }''' % (_FIELDS, _EXTRA_FIELDS),
+}
+
+
+class AllstarBaseIE(InfoExtractor):
+    @staticmethod
+    def _parse_video_data(video_data):
+        def _media_url_or_none(path):
+            return urljoin('https://media.allstar.gg/', path)
+
+        def _profile_url_or_none(path):
+            return urljoin('https://allstar.gg/u/', path)
+
+        return traverse_obj(video_data, {
+            'id': ('_id', {str}),
+            'display_id': ('shareId', {str}),
+            'title': ('clipTitle', {str}),
+            'url': ('clipLink', {_media_url_or_none}),
+            'thumbnail': ('clipImageThumb', {_media_url_or_none}),
+            'duration': ('clipLength', {int_or_none}),
+            'filesize': ('clipSizeBytes', {int_or_none}),
+            'timestamp': ('createdDate', {int_or_none}),
+            'uploader': ('username', {str}),
+            'uploader_id': ('user', '_id', {str}),
+            'uploader_url': ('user', '_id', {_profile_url_or_none}),
+            'view_count': ('views', {int_or_none}),
+        })
```

Comment:
ok done.

I have a quation.
some videos have hls fields:
- `clipLinkHLS` return `404 error` most of the time.
- `cf { hls dash }` they return the same video/audio tracks, but this fields not allways present.
should I use theme to extract formats or just keep it simple and stick to direct mp4 url?

Reply:
If the HLS link returns 404 most of the time and doesn't offer any higher quality formats, then I'd say it's not worth it. It would just create more network overhead, especially during playlist extraction

Reply:
exmaple of available formats of `cf hls` for `clip=6489a71bb4ae6f1f0f78cb35`:
```
group_audio-original m3u8 audio only     │                 m3u8  │ audio only        unknown    [en-371c902d] original
540                  mp4  426x240     60 │ ~ 1.58MiB  541k m3u8  │ avc1.42c01e  541k video only
883                  mp4  640x360     60 │ ~ 2.59MiB  883k m3u8  │ avc1.4d401f  883k video only
1384                 mp4  854x480     60 │ ~ 4.06MiB 1385k m3u8  │ avc1.4d401f 1385k video only
2595                 mp4  1280x720    60 │ ~ 7.60MiB 2596k m3u8  │ avc1.4d4020 2596k video only
4962                 mp4  1920x1080   60 │ ~14.54MiB 4963k m3u8  │ avc1.4d402a 4963k video only
```
and the direct mp4 `url`:
```
Video: h264 (Main) (avc1 / 0x31637661), yuv420p(progressive), 1920x1080 [SAR 1:1 DAR 16:9], 13055 kb/s, 60 fps, 60 tbr, 15360 tbn
Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 135 kb/s
```
the `cf hls` just provide separated autio/video tracks at lower qualities.
so as you said it's not worth it, lets keep it simple.